wave_benchmark/src/trajectory_compare.cpp

```diff
@@ -0,0 +1,44 @@
+#include "wave/benchmark/trajectory_compare.hpp"
+
+namespace wave {
+
+void TrajectoryCompare::reset() {
+    this->ground_truth.clear();
+    this->measurements.clear();
+    this->error.clear();
+}
+
+void TrajectoryCompare::push_measurement(Pose pose, const TimeType time) {
+    measurements.insert(PoseMeasurement(time, this->measurement_key, pose));
+}
+
+void TrajectoryCompare::push_truth(Pose pose, const TimeType time) {
+    ground_truth.insert(PoseMeasurement(time, this->ground_truth_key, pose));
+}
+
+void TrajectoryCompare::calculate_error() {
```

Comment:
That's a good point. I will add that but I think I'm going to keep the one-by-one feeding just because it is convenient if the measurements are coming from a rosbag: The rosbag API isn't well documented and the only way I know to consume them is to iterate through, so 1-by-1 feeding is natural.